k2/csrc/fsa_algo_test.cu

```diff
@@ -585,4 +585,119 @@ TEST(FsaAlgo, Determinize) {
   }
 }
 
+TEST(FsaAlgo, ClosureSimpleCase) {
+  // 0 -> 1 -> 2 -> 3
+  std::string s = R"(0 1 1 0.1
+    1 2 2 0.2
+    2 3 -1 0.3
+    3
+  )";
+
+  for (auto &context : {GetCpuContext(), GetCudaContext()}) {
+    Fsa fsa = FsaFromString(s).To(context);
+    Array1<int32_t> arc_map;
+    Fsa ans = Closure(fsa, &arc_map);
+
+    arc_map = arc_map.To(GetCpuContext());
+    CheckArrayData(arc_map, std::vector<int32_t>{0, -1, 1, 2});
+
+    ans = ans.To(GetCpuContext());
+    EXPECT_EQ(ans.TotSize(0), 4);  // number of states
+    EXPECT_EQ(ans.TotSize(1), 4);  // number of arcs
+
+    EXPECT_EQ((ans[{0, 0}]), (Arc{0, 1, 1, 0.1f}));   // state 0, arc 0
```

Comment:
Just metion it (we don't need to do this now), the current test approach depends on the implementation, we may need to find a implementation independent way to test `Closure`. One of a way I think of is, we construct a src FSA with every leaving arc for a state has different label, then for any path `A` in src, check if we can find AA in ans, or check if we can find A in ans and the last state has an arc entering the start_state with label 0, etc. Not sure if @danpovey has any better idea.